rust/agama-server/src/network/web.rs

```diff
@@ -167,67 +175,142 @@ async fn connections(State(state): State<NetworkState>) -> Json<Vec<NetworkConne
 async fn add_connection(
     State(state): State<NetworkState>,
     Json(conn): Json<NetworkConnection>,
-) -> Result<Json<Uuid>, NetworkError> {
+) -> Result<Json<Connection>, NetworkError> {
     let (tx, rx) = oneshot::channel();
 
+    let conn = Connection::try_from(conn)?;
+    let id = conn.id.clone();
+
     state
         .actions
-        .send(Action::AddConnection(
-            conn.id.clone(),
-            conn.device_type(),
-            tx,
-        ))
+        .send(Action::NewConnection(Box::new(conn.clone()), tx))
         .unwrap();
     let _ = rx.await.unwrap();
 
-    let conn = Connection::try_from(conn)?;
-    let uuid = conn.uuid.clone();
-
+    let (tx, rx) = oneshot::channel();
     state
         .actions
-        .send(Action::UpdateConnection(Box::new(conn)))
+        .send(Action::GetConnection(id.clone(), tx))
         .unwrap();
 
-    Ok(Json(uuid))
+    match rx.await.unwrap() {
+        None => Err(NetworkError::CannotAddConnection(id.clone())),
+        Some(conn) => Ok(Json(conn)),
+    }
 }
 
-#[utoipa::path(delete, path = "/network/connections/:uuid", responses(
+#[utoipa::path(delete, path = "/network/connections/:id", responses(
   (status = 200, description = "Delete connection", body = Connection)
 ))]
 async fn delete_connection(
     State(state): State<NetworkState>,
-    Path(id): Path<Uuid>,
+    Path(id): Path<String>,
 ) -> Result<Json<()>, NetworkError> {
     let (tx, rx) = oneshot::channel();
     state
         .actions
         .send(Action::RemoveConnection(id, tx))
         .unwrap();
 
-    let _ = rx.await.unwrap();
-    let (tx, rx) = oneshot::channel();
-    state.actions.send(Action::Apply(tx)).unwrap();
-    let _ = rx.await.unwrap();
-
-    Ok(Json(()))
+    Ok(Json(rx.await.unwrap()?))
 }
 
-#[utoipa::path(put, path = "/network/connections/:uuid", responses(
+#[utoipa::path(put, path = "/network/connections/:id", responses(
   (status = 200, description = "Update connection", body = Connection)
 ))]
 async fn update_connection(
     State(state): State<NetworkState>,
-    Path(id): Path<Uuid>,
+    Path(id): Path<String>,
     Json(conn): Json<NetworkConnection>,
 ) -> Result<Json<()>, NetworkError> {
+    let (tx, rx) = oneshot::channel();
+    state
+        .actions
+        .send(Action::GetConnection(id.clone(), tx))
+        .unwrap();
+    let current_conn = rx.await.unwrap();
```

Comment:
What about using `old_conn` or `orig_conn`? Or use `conn` here and rename `conn` to `updated_conn`.